server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -458,15 +433,27 @@ private void checkSystemIndexAccess(Context context, Set<Index> concreteIndices)
         }
     }
 
+    private void ensureRemoteIndicesRequireIgnoreUnavailable(IndicesOptions options, String... indexExpressions) {
+        if (options.ignoreUnavailable() == false && indexExpressions != null) {
+            List<String> crossClusterIndices = Arrays.stream(indexExpressions).filter(index -> index.contains(":")).toList();
+            if (crossClusterIndices.size() > 0) {
+                throw new IllegalArgumentException(
+                    "Cross-cluster calls are not supported in this context but remote indices were requested: " + crossClusterIndices
+                );
+            }
+        }
+    }
+
     private static IndexNotFoundException notFoundException(String... indexExpressions) {
-        IndexNotFoundException infe;
-        if (indexExpressions != null && indexExpressions.length == 1) {
-            if (Metadata.ALL.equals(indexExpressions[0])) {
-                infe = new IndexNotFoundException("no indices exist", indexExpressions[0]);
+        final IndexNotFoundException infe;
+        if (indexExpressions == null || indexExpressions.length <= 1) {
```

Comment:
I feel like these clauses could be simplified - eg you're checking for `indexExpressions == null` twice

Reply:
Good point! pushed [154f450](https://github.com/elastic/elasticsearch/pull/91735/commits/154f45016ed2596574d9ea2c63a020d693305e98)